model/jpa/src/main/java/org/keycloak/models/jpa/RoleAdapter.java

```diff
@@ -89,30 +96,88 @@ public void setName(String name) {
 
     @Override
     public boolean isComposite() {
-        return getCompositesStream().count() > 0;
+        TypedQuery<String> query = em.createNamedQuery("getChildrenRoleIds", String.class);
+        query.setParameter("roleId", getId());
+        query.setMaxResults(1);
+        return !query.getResultList().isEmpty();
```

Comment:
When the composite roles have been initialized, using the collection might be faster than triggering another query. 
Therefore, re-use the same pattern `Hibernate.isPropertyInitialized(getEntity(), "compositeRoles")` here as well.

Reply:
Absolutely. Will do.

Reply:
Done.
Test `ClientRegistrationTest` passing with Undertow, Wildfly and Quarkus.